lib/tasks/multitenant_db_merge.rake

```diff
@@ -0,0 +1,344 @@
+           })
+    res['patient_id'].nil? ? nil : res
+  }
+  @note_mappings = easy_mass_insert Note, 'notes', note_map, false
+
+  # Port call list entries
+  cle_map = -> (x) {
+    res = x.except('id', 'created_at', 'updated_at', 'fund_id', 'line_id', 'patient_id', 'user_id')
+           .merge({
+             'fund_id' => @fund_id,
+             'created_at' => x['created_at'].asctime.in_time_zone("America/New_York"),
+             'updated_at' => x['updated_at'].asctime.in_time_zone("America/New_York"),
+             'line_id' => @line_mappings[x['line_id']],
+             'patient_id' => @patient_mappings[x['patient_id']],
+             'user_id' => @user_mappings[x['user_id']]
+           })
+    res['patient_id'].nil? ? nil : res
+  }
+  @call_list_entry_mappings = easy_mass_insert CallListEntry, 'call_list_entries', cle_map, false
+
+  # Port events
+  event_map = -> (x) {
+    res = x.except('id', 'created_at', 'updated_at', 'fund_id', 'line_id', 'patient_id')
+           .merge({
+             'fund_id' => @fund_id,
+             'created_at' => x['created_at'].asctime.in_time_zone("America/New_York"),
+             'updated_at' => x['updated_at'].asctime.in_time_zone("America/New_York"),
+             'line_id' => @line_mappings[x['line_id']],
+             'patient_id' => @patient_mappings[x['patient_id'].to_i],
+           })
+    res['patient_id'].nil? ? nil : res
+  }
+  @event_mappings = easy_mass_insert Event, 'events', event_map, false
+
+  # Versions, maybe a little more complicated...
+  @fkey_mappings = {
+    'fund_id' => {@fund_id => @fund_id},
+    'clinic_id' => @clinic_mappings,
+    'pledge_generated_by_id' => @user_mappings,
+    'pledge_sent_by_id' => @user_mappings,
+    'last_edited_by_id' => @user_mappings,
+    'user_id' => @user_mappings,
+    'line_id' => @line_mappings,
+    'patient_id' => @patient_mappings,
+    'can_call_id' => @patient_mappings,
+    'can_pledge_id' => @patient_mappings,
+    'can_fulfill_id' => @patient_mappings,
+    'can_support_id' => @patient_mappings
+  }
+  @item_type_mappings = {
+    'ArchivedPatient' => @archived_patient_mappings,
+    'Call' => @call_mappings,
+    'CallListEntry' => @call_list_entry_mappings,
+    'Clinic' => @clinic_mappings,
+    'Config' => @config_mappings,
+    'ExternalPledge' => @external_pledge_mappings,
+    'Event' => @event_mappings,
+    'Fulfillment' => @fulfillment_mappings,
+    'Line' => @line_mappings,
+    'Note' => @note_mappings,
+    'Patient' => @patient_mappings,
+    'User' => @user_mappings,
+    'PracticalSupport' => @practical_support_mappings,
+  }
+
+  def transform_obj(obj, item_type, value_will_be_array)
+    obj.each_pair do |k, v|
+      obj['id'] = @item_type_mappings[item_type][v] if k == 'id'
+      if @fkey_mappings.keys.include? k || (k.start_with?('can_') && k.end_with?('_id'))
+        keyset = if k == 'can_call_id'
+                   obj['can_call_type'] == 'Patient' ? @patient_mappings : @archived_patient_mappings
```

Comment:
Why do we trust this to be either/or now, unlike up in the subobjects? are we more certain we won't find any abandoned ones here?

Reply:
nah, def possible that some of these will be abandoned, but the versions table is storing everything as json rather than references like on rails models, so it won't raise errors if it runs into an abandoned obj. In other words, here we're just manipulating numbers in json rather than ensuring foreign key relations are valid, so we can care a lot less.